### ISSUE TITLE
<noscript> tags when JS enabled are not ignored

### ISSUE THREAD

**Issue by snewcomer** [glimmerjs/glimmer-vm#839]
@jsit and I found a weird bug.

When placing an `img` or `style` tag in `<noscript>` tags, the content inside still gets applied when JS is enabled.  I found this in a fastboot (ember 3.1.2) app as well as in a non-fastboot app (ember 3.3).

A simple test to try in application.hbs is (with a variety of scenarios depending on your app setup):

```
<noscript>
  <style>
    body {
      background-color: red !important;
      font-size: 0.2em !important;
      
      > .ember-view {
         background-color: red !important;
      }
    }
  </style>
</noscript>
```

Let me know if anybody sees something different!

**Comment by chancancode** [comment]
I agree this is a bug that should probably be fixed, though, in practice I think it's probably best to put this in your `index.html` anyway. Are you using server-side-rendering?

**Comment by snewcomer** [comment]
Yep, a server side rendered app.

Also, this use case is for lazy loading images (in a component).  When JS is disabled, we still want to render the original high res image served from fastboot.  But when enabled, we would like to save some bandwidth; however, seeing the unexpected results.

**Comment by chancancode** [comment]
Makes sense. I think `<noscript>`...

* should be rendered in SSR
* should be removed during rehydration
* should not be rendered at client-side render

**Comment by rwjblue** [comment]
I guess I don't grok why the browser doesn't just take care of this for us?

**Comment by rwjblue** [comment]
Anyone have a link handy to the specs around `noscript`?

**Comment by chancancode** [comment]
@rwjblue I suppose we can file a bug, but `document.createElement('noscript')` is quite nonsensical. I suspect the `noscript` semantics might just all be a parse time/tree-construction time thing.

**Comment by chancancode** [comment]
https://html.spec.whatwg.org/multipage/scripting.html#the-noscript-element

So, a couple of problems:

> The noscript element represents nothing if scripting is enabled, and represents its children if scripting is disabled. It is used to present different markup to user agents that support scripting and those that don't support scripting, by affecting how the document is **parsed**.

But on the other hand: 

> The noscript element has no other requirements. In particular, children of the noscript element are not exempt from form submission, scripting, and so forth, even when scripting is enabled for the element.

So, omitting them on CSR is not exactly right either, though probably no one would care.

**Comment by rwjblue** [comment]
I'm just not sure that we have a bug here. It seems to me that folks should guard any `<noscript>` content in an `{{#if isFastBoot}}` or somesuch.  As far as I can tell rendering `<noscript>` content _only_ makes sense during SSR...

**Comment by snewcomer** [comment]
`{{isFastBoot}}` seems like a good solution for fastboot enabled apps!  Thanks.  My only remaining question would be for non fastboot apps.  I haven't found a solution yet.

**Comment by rwjblue** [comment]
Hmm, I don’t understand what you mean. If an app doesn’t have fastboot, you never render without JavaScript and therefore should never have `<noscript>`. What am I missing?

**Comment by snewcomer** [comment]
I'm sorry.  Yes you are correct.  I wouldn't put noscripts in there but for the index.html.  

I think this potentially could be closed since the fastboot check solved the issue for me.

**Comment by chancancode** [comment]
> In the React world, this is a common pattern for dealing with lazy loaded images—user agents with scripting get the empty placeholder, user agents without scripting get the normal <img> tag.

@tomdale I'm not sure if I understand how this works. Can you elaborate with some example markup/JSX?

**Comment by chancancode** [comment]
Reading the spec again, I think I misunderstood it. When scripting is enabled, what we are supposed to do is to insert the content as text.
